autolab/Autolab
# Lint views/announcements and Touch-up UI

## Summary

### Summary generated by Reviewpad on 06 Oct 23 17:35 UTC
This pull request includes two patches. The first patch fixes linting issues and updates the user interface in the views/announcements directory. The second patch addresses some minor issues in the app/views/announcements/index.html.erb file.


## Description

- Run `erblint` on `views/announcements` using `bundle exec erblint app/views/announcements/*.html.erb -a`      
- Complete a TODO in `_announcementFields.html.erb` to add better titles to checkboxes
- Update edit and create views' containers to fill up page
- Make index display "There are no existing announcements." if no announcements, turn "Create Announcement" into a styled button, and remove "Save All Changes" (this just linked to the course page)

## Motivation and Context


- as part of summer linting

## How Has This Been Tested?




- As instructor, Go to "Manage Course", go to "Manage announcements", see that announcements page renders correctly, any existing announcements are displayed
- Create and edit an announcement successfully
- View announcement as a student

### Create page

Before: 

<img width="1512" alt="Screenshot 2023-08-14 at 10 24 39 AM" src="https://github.com/autolab/Autolab/assets/25730111/70806248-0794-4931-93ad-d124cb814b72">

After:

<img width="1475" alt="Screenshot 2023-08-11 at 5 22 21 PM" src="https://github.com/autolab/Autolab/assets/25730111/02e90cf6-7475-4c70-8584-ddb1e8dcc15a">

### Edit page

Before:

<img width="1496" alt="Screenshot 2023-08-14 at 10 24 47 AM" src="https://github.com/autolab/Autolab/assets/25730111/909704d7-61b7-4754-ba44-5db892da38b0">

After:

<img width="1042" alt="Screenshot 2023-08-11 at 5 21 28 PM" src="https://github.com/autolab/Autolab/assets/25730111/3126ea5e-8de0-49a9-8c64-bfdb0a8fa5f9">

### Index page

Before:

<img width="1512" alt="Screenshot 2023-08-14 at 10 24 19 AM" src="https://github.com/autolab/Autolab/assets/25730111/8a57ee49-1c54-414d-8137-347158067262">


After:

<img width="1512" alt="Screenshot 2023-08-11 at 5 21 53 PM" src="https://github.com/autolab/Autolab/assets/25730111/705dbd9f-8664-431b-8118-7dedcedb55ee">

<img width="1512" alt="Screenshot 2023-08-11 at 5 22 52 PM" src="https://github.com/autolab/Autolab/assets/25730111/8e9f0352-c9c6-41fb-978e-5e82e5e170e9">


## Checklist:


## Other issues / help required




If unsure, feel free to submit first and we'll help you along.
